pkg/state/degraphql_route_test.go

```diff
@@ -0,0 +1,278 @@
+			}
+		})
+	}
+}
+
+func TestDegraphqlRouteGet(t *testing.T) {
+	assert := assert.New(t)
+	collection := degraphqlRoutesCollection()
+
+	var degraphqlRoute DegraphqlRoute
+	degraphqlRoute.ID = kong.String("example")
+	degraphqlRoute.URI = kong.String("/foo")
+	degraphqlRoute.Query = kong.String("query { hello }")
+	degraphqlRoute.Service = &kong.Service{
+		ID: kong.String("some-service"),
+	}
+	degraphqlRoute.Methods = kong.StringSlice("GET")
+
+	err := collection.Add(degraphqlRoute)
+	assert.Nil(err)
+
+	// Fetch the currently added entity
+	res, err := collection.Get("example")
+	assert.Nil(err)
+	assert.NotNil(res)
+	assert.Equal("example", *res.ID)
+	assert.Equal("some-service", *res.Service.ID)
+
+	// Fetch non-existent entity
+	res, err = collection.Get("does-not-exist")
+	assert.NotNil(err)
+	assert.Nil(res)
+}
+
+func TestDegraphqlRouteUpdate(t *testing.T) {
+	assert := assert.New(t)
+	collection := degraphqlRoutesCollection()
+
+	var degraphqlRoute DegraphqlRoute
+	degraphqlRoute.ID = kong.String("example")
+	degraphqlRoute.URI = kong.String("/foo")
+	degraphqlRoute.Query = kong.String("query { hello }")
+	degraphqlRoute.Service = &kong.Service{
+		ID: kong.String("some-service"),
+	}
+	degraphqlRoute.Methods = kong.StringSlice("GET")
+
+	err := collection.Add(degraphqlRoute)
+	assert.Nil(err)
+
+	// Fetch the currently added entity
+	res, err := collection.Get("example")
+	assert.Nil(err)
+	assert.NotNil(res)
+	assert.Equal("query { hello }", *res.Query)
+
+	// Update query field
+	res.Query = kong.String("query { hello world }")
+	err = collection.Update(*res)
+	assert.Nil(err)
+
+	// Fetch again
+	res, err = collection.Get("example")
+	assert.Nil(err)
+	assert.NotNil(res)
+	assert.Equal("query { hello world }", *res.Query)
+}
+
+func TestDegraphqlRouteDelete(t *testing.T) {
+	assert := assert.New(t)
+	collection := degraphqlRoutesCollection()
+
+	var degraphqlRoute DegraphqlRoute
+	degraphqlRoute.ID = kong.String("example")
+	degraphqlRoute.URI = kong.String("/foo")
+	degraphqlRoute.Query = kong.String("query { hello }")
+	degraphqlRoute.Service = &kong.Service{
+		ID: kong.String("some-service"),
+	}
+	degraphqlRoute.Methods = kong.StringSlice("GET")
+
+	err := collection.Add(degraphqlRoute)
+	assert.Nil(err)
+
+	// Fetch the currently added entity
+	res, err := collection.Get("example")
+	assert.Nil(err)
+	assert.NotNil(res)
+
+	// Delete entity
+	err = collection.Delete(*res.ID)
+	assert.Nil(err)
+
+	// Fetch again
+	res, err = collection.Get("example")
+	assert.NotNil(err)
+	assert.Nil(res)
+}
+
+func TestDegraphqlRouteGetAll(t *testing.T) {
+	assert := assert.New(t)
+	collection := degraphqlRoutesCollection()
+
+	populateDegraphqlRoutes(assert, collection)
+
+	degraphqlRoutes, err := collection.GetAll()
+	assert.Nil(err)
+	assert.Equal(5, len(degraphqlRoutes))
+	assert.IsType([]*DegraphqlRoute{}, degraphqlRoutes)
+}
+
+func populateDegraphqlRoutes(assert *assert.Assertions,
```

Comment:
Nit: this is fine but I'd consider passing the `t *testing.T` instead of the `assert` object as using the latter can potentially (not necessarily) result in an unrelated `assert` being used.

IMHO the former is also more idiomatic. WDYT?